crates/teloxide/src/dispatching/tracing.rs

```diff
@@ -0,0 +1,34 @@
+use super::{DpHandlerDescription, UpdateHandler};
+
+use dptree::{di::Injectable, prelude::DependencyMap, HandlerDescription};
+use tracing::{Instrument, Span};
+
+/// Extension methods for adding `tracing` to [`UpdateHandler`].
+///
+/// This trait is only available if the `tracing` feature is enabled.
+pub trait UpdateHandlerTracingExt<E> {
+    /// Returns an `UpdateHandler` with tracing enabled.
```

Comment:
You can't disable tracing... This doesn't enable it, this just wraps everything in a span.